lib/_http_server.js

```diff
@@ -217,6 +217,11 @@ function writeHead(statusCode, reason, obj) {
     // This class of status code indicates a provisional response,
     // consisting only of the Status-Line and optional headers, and is
     // terminated by an empty line.
+
+    if(statusCode == 304 && (headers.hasOwnProperty('trailer') || headers.hasOwnProperty('Trailer')){
```

Comment:
Ugh, yes, you're right. Forgot about that. The inconsistency on that bit is irritating. Good suggestion on moving the check to `_storeHeader()`